mne/cov.py

```diff
@@ -760,10 +759,32 @@ def _compute_covariance_auto(data, method, info, method_params, cv,
     return out
 
 
+def _logdet(A):
+    vals = linalg.eigh(A)[0]
+    vals = np.abs(vals)  # avoid negative values (numerical errors)
```

Comment:
@Eric89GXL that's the culprit. eigh returns negative eig vals on a rank deficient symmetric matrix.